src/tree_data.h

```diff
@@ -997,6 +997,20 @@ struct lyd_node_opaq {
     const struct ly_ctx *ctx;       /**< libyang context */
 };
 
+/**
+ * @brief Context term data node record.
+ */
+struct lyd_term_nodes_ext_rec {
+    const struct lyd_node_term *node;           /** pointer to the node itself */
+    const struct lyd_node_term **leafref_nodes; /** list of the leafref data nodes [sized array](@ref sizedarrays)).
+                                                    By default it is empty. It is filled automatically by validation
+                                                    function of leafref nodes. It can also be populated based on manual request
+                                                    using [link api](@ref lyd_link_leafref_node_tree). Freeing of the resources
+                                                    is automatic. */
+    const struct lyd_node_term *target_node;    /** pointer to leafref target data node, by default is NULL. The logic
```

Comment:
Mention that it is filled only for leafrefs.

Reply:
Done